routing/payment_lifecycle.go

```diff
@@ -38,21 +38,53 @@ type paymentState struct {
 	numShardsInFlight int
 	remainingAmt      lnwire.MilliSatoshi
 	remainingFees     lnwire.MilliSatoshi
-	terminate         bool
+
```

Comment:
is this commit really needed now? How is it making the code more testable?

Reply:
Aha, I see unit tests added for the payment state in later commits. Maybe move those commits together to amke it more obvious why this change is done?

Reply:
Yep I've moved them all into one commit.